gplugins/design_recipe/DesignRecipe.py

```diff
@@ -25,57 +27,64 @@ class DesignRecipe:
     # the hash of the system last time eval() was executed
     last_hash: int
 
-    # The component this DesignRecipe operates on. This is not necessarily
-    # the same `component` refered to in the `dependencies` recipes.
-    component: Optional[Component] = None
+    # The component factory this DesignRecipe operates on. This is not necessarily
+    # the same `component` referred to in the `dependencies` recipes.
+    cell: ComponentFactory | Component | None = None
 
     # LayerStack for the process that the component is generated for
     layer_stack: LayerStack
 
-    def __init__(self,
-                 component: Component,
-                 layer_stack: LayerStack = get_layer_stack(),
-                 dependencies: List[dr.DesignRecipe] = []):
+    def __init__(
+        self,
+        cell: ComponentFactory | Component,
+        dependencies: list[dr.DesignRecipe] | None = None,
+        layer_stack: LayerStack = get_layer_stack(),
+    ):
+        dependencies = dependencies or []
         self.dependencies = dr.ConstituentRecipes(dependencies)
-        self.component = component
+        self.cell = cell
         self.last_hash = -1
         self.layer_stack = layer_stack
 
     def __hash__(self) -> int:
         """
         Returns a hash of all state this DesignRecipe contains.
-        Subclasses should include functionality-specific state (e.g. fdtd settings) here. 
+        Subclasses should include functionality-specific state (e.g. fdtd settings) here.
         This is used to determine 'freshness' of a recipe (i.e. if it needs to be rerun)
+
+        Hashed items:
+        - component or cell
```

Comment:
clarify that it's the component/cell geometry not the whole thing (which might include non-static state)